### PR TITLE
Feature/branch name

### PR DESCRIPTION
**Pull request by marmoure**
Adds the option of using a different branch as base instead of `main`.

# how to test this PR:
* from a clean state (remove node_modules & any fork) 
* checkout the branch
* install dependencies
* start the demo `yarn start:demo`
* open `http://localhost:1234/?ghrepo=evolvedbinary/prosemirror-lwdita&source=packages/prosemirror-lwdita-demo/example-xdita/02-short-file.xml&branch=ignore/link&referer=htwsrhtshrts`
* the short-example.xml should reflect the current state of the one in the branch `ignore/link`
* Create a new branch on `prosemirror-lwdita`
* change the state of one of the example file
* make a request to `http://localhost:1234/?ghrepo=evolvedbinary/prosemirror-lwdita&source=packages/prosemirror-lwdita-demo/example-xdita/02-short-file.xml&branch=ignore/link&referer=htwsrhtshrts`, replace the `ignore/link` branch with your new branch
* the short-example.xml should reflect the current state of the one in your new branch
* start the backend `yarn start:backend`
* authorize and install petal GitHub app
* make a request to `http://localhost:1234/?ghrepo=evolvedbinary/prosemirror-lwdita&source=packages/prosemirror-lwdita-demo/example-xdita/02-short-file.xml&branch=ignore/link&referer=htwsrhtshrts`
* edit the file 
* click publish
* fill out the form
* check the create pr, it should be created against the base `ignore/link`

### CODE REVIEW

File: packages/prosemirror-lwdita/src/github-integration/github.plugin.ts [hunk 1]
```diff
@@ -113,13 +113,13 @@ export const getUserInfo = async (token: string): Promise<Record<string, string>
  * @param changedDocument - The content of the changed document.
  * @returns A promise that resolves when the document has been published.
  */
-export const createPrFromContribution = async (ghrepo: string, source: string, changedDocument: string, title: string, desc: string): Promise<string> => {
+export const createPrFromContribution = async (ghrepo: string, source: string, branch: string, changedDocument: string, title: string, desc: string): Promise<string> => {
```
Comment (adamretter) [inline packages/prosemirror-lwdita/src/github-integration/github.plugin.ts:116]: tsdoc params need updating.

File: packages/prosemirror-lwdita/src/github-integration/github.plugin.ts [hunk 2]
```diff
@@ -29,9 +29,9 @@ import { document as jditaToProsemirrorJson } from "../document";
  * This function currently fetches the document from the 'main' branch of the repository.
  * should use the GitHub API to dynamically determine the default branch of the repository.
  */
-export const fetchRawDocumentFromGitHub = async (ghrepo: string, source: string): Promise<string> => {
+export const fetchRawDocumentFromGitHub = async (ghrepo: string, source: string, branch: string): Promise<string> => {
```
Comment (adamretter) [inline packages/prosemirror-lwdita/src/github-integration/github.plugin.ts:32]: tsdoc params need updating.

File: packages/prosemirror-lwdita/src/commands.ts [hunk 3]
```diff
@@ -181,7 +181,7 @@ export class InputContainer {
 /**
  * Render a dialog form for inserting the PR metadata
  */
-export function renderPrDialog(ghrepo: string, source: string, updatedXdita: string): void {
+export function renderPrDialog(ghrepo: string, source: string, branch: string, updatedXdita: string): void {
```
Comment (adamretter) [inline packages/prosemirror-lwdita/src/commands.ts:184]: tsdoc is missing params

File: packages/prosemirror-lwdita-backend/src/api/modules/octokit.module.mts [hunk 4]
```diff
@@ -123,24 +122,19 @@ const createFork = async (octokit: Octokit, owner: string, repo: string): Promis
  * @param newBranch - The name of the new branch to be created.
  * @returns A Promise that resolves to a BranchInfo object representing the newly created branch, or undefined if an error occurs.
  */
-const createBranch = async (octokit: Octokit, owner: string, repo: string, newBranch: string): Promise<BranchInfo | undefined> => {
+const createBranch = async (octokit: Octokit, owner: string, repo: string, branch: string, newBranch: string): Promise<BranchInfo | undefined> => {
```
Comment (adamretter) [inline packages/prosemirror-lwdita-backend/src/api/modules/octokit.module.mts:125]: tsdoc params need updating.

File: packages/prosemirror-lwdita/src/github-integration/github.plugin.ts [hunk 5]
```diff
@@ -113,13 +113,13 @@ export const getUserInfo = async (token: string): Promise<Record<string, string>
...
  */
-export const createPrFromContribution = async (ghrepo: string, source: string, changedDocument: string, title: string, desc: string): Promise<string> => {
+export const createPrFromContribution = async (ghrepo: string, source: string, branch: string, changedDocument: string, title: string, desc: string): Promise<string> => {
   const authenticatedUserInfo = await getUserInfo(localStorage.getItem('token') as string);
   
   const owner = ghrepo.split('/')[0];
   const repo = ghrepo.split('/')[1];
   const newOwner = authenticatedUserInfo.login;
-  const newBranch = "new-branch-petal-app";
+  const newBranch = "new-branch-petal";
```
Comment (adamretter) [inline packages/prosemirror-lwdita/src/github-integration/github.plugin.ts:122]: What is this `new-branch-petal` used for?

File: packages/prosemirror-lwdita/src/github-integration/request.ts [hunk 6]
```diff
@@ -32,7 +32,7 @@ export interface URLParams {
  * source = GitHub resource,
  * referer = Referer of the request
  */
-export const validKeys = ['ghrepo', 'source', 'referer'];
+export const validKeys = ['ghrepo', 'source', 'branch', 'referer'];
```
Comment (adamretter) [inline packages/prosemirror-lwdita/src/github-integration/request.ts:35]: tsdoc needs updating.

**Review by adamretter — CHANGES_REQUESTED** [review]
Needs some small changes or tsdoc additions

**Comment by marmoure** [comment]
> Needs some small changes or tsdoc additions

I suppose the linter did not catch these  tsdoc issues, I will add that to the backlog.

File: packages/prosemirror-lwdita/src/github-integration/github.plugin.ts (hunk 5 above)
Comment (marmoure) [reply packages/prosemirror-lwdita/src/github-integration/github.plugin.ts:122]: This is a new branch created by the GitHub App to push changes to, and create the PR from.

File: packages/prosemirror-lwdita/src/github-integration/request.ts (hunk 6 above)
Comment (marmoure) [reply packages/prosemirror-lwdita/src/github-integration/request.ts:36]: Fixed in b6962288f2f38f863a94da06d3e6f53b77fe42fa .

File: packages/prosemirror-lwdita-backend/src/api/modules/octokit.module.mts (hunk 4 above)
Comment (marmoure) [reply packages/prosemirror-lwdita-backend/src/api/modules/octokit.module.mts:126]: Fixed in b6962288f2f38f863a94da06d3e6f53b77fe42fa .

File: packages/prosemirror-lwdita-backend/src/api/modules/octokit.module.mts [hunk 7]
```diff
@@ -282,29 +276,28 @@ const createPullRequest = async (octokit: Octokit, owner: string, repo: string,
  * @param body - The body of the pull request.
  * @returns The URL of the created pull request, or undefined if an error occurred.
  */
-export const pushChangesAndCreatePullRequest = async (octokit: Octokit, owner: string, repo: string, newOwner:string, newBranch: string, commitMessage: string, change: { path: string, content: string }, title: string, body: string): Promise<string | undefined> => {
+export const pushChangesAndCreatePullRequest = async (octokit: Octokit, owner: string, repo: string, newOwner:string, branch: string, newBranch: string, commitMessage: string, change: { path: string, content: string }, title: string, body: string): Promise<string | undefined> => {
```
[the remark this answers is not in the document]
Comment (marmoure) [reply packages/prosemirror-lwdita-backend/src/api/modules/octokit.module.mts:281]: Fixed in b6962288f2f38f863a94da06d3e6f53b77fe42fa .

File: packages/prosemirror-lwdita/src/commands.ts (hunk 3 above)
Comment (marmoure) [reply packages/prosemirror-lwdita/src/commands.ts:189]: Fixed in b6962288f2f38f863a94da06d3e6f53b77fe42fa .

File: packages/prosemirror-lwdita/src/github-integration/github.plugin.ts [hunk 8]
```diff
@@ -62,8 +62,8 @@ export const transformGitHubDocumentToProsemirrorJson = async (rawDocument: stri
  * @param source - The source path of the document within the repository.
  * @returns A promise that resolves to the transformed ProseMirror JSON document.
  */
-export const fetchAndTransform = async (ghrepo: string, source: string) => {
-  const rawDoc = await fetchRawDocumentFromGitHub(ghrepo, source);
+export const fetchAndTransform = async (ghrepo: string, source: string, branch: string) => {
```
[the remark this answers is not in the document]
Comment (marmoure) [reply packages/prosemirror-lwdita/src/github-integration/github.plugin.ts:66]: Fixed in b6962288f2f38f863a94da06d3e6f53b77fe42fa .

File: packages/prosemirror-lwdita/src/github-integration/github.plugin.ts (hunk 1 above)
Comment (marmoure) [reply packages/prosemirror-lwdita/src/github-integration/github.plugin.ts:118]: Fixed in b6962288f2f38f863a94da06d3e6f53b77fe42fa .

File: packages/prosemirror-lwdita/src/github-integration/github.plugin.ts [hunk 9]
```diff
@@ -29,9 +29,9 @@ import { document as jditaToProsemirrorJson } from "../document";
  * This function currently fetches the document from the 'main' branch of the repository.
  * should use the GitHub API to dynamically determine the default branch of the repository.
  */
-export const fetchRawDocumentFromGitHub = async (ghrepo: string, source: string): Promise<string> => {
+export const fetchRawDocumentFromGitHub = async (ghrepo: string, source: string, branch: string): Promise<string> => {
   //TODO(YB): the branch should be passed as a parameter
```
[the remark this answers is not in the document]
Comment (marmoure) [reply packages/prosemirror-lwdita/src/github-integration/github.plugin.ts]: Fixed in b6962288f2f38f863a94da06d3e6f53b77fe42fa .

File: packages/prosemirror-lwdita/src/github-integration/github.plugin.ts (hunk 5 above)
Comment (adamretter) [reply packages/prosemirror-lwdita/src/github-integration/github.plugin.ts:124]: Okay interesting. Then I think we need to improve on this. Especially as they might want to use Petal to edit more than one page in the same repo, and if we hardcode the branch name, we will get a conflict when they edit a second page. Perhaps this should be something like `doc/petal-${timestamp}` where the timestamp is a serial timestamp like `YYYYMMDDHHmmSS` (on the UTC timezone)

Comment (marmoure) [reply packages/prosemirror-lwdita/src/github-integration/github.plugin.ts]: Fixed in 58516fd2a46ddf0292097ac7cfa6e33bb293725d

**Pull request merged**